users/views.py

```diff
@@ -137,3 +140,70 @@ def post(self, request):
             return JsonResponse({'MESSAGE': 'KEY_ERROR'}, status=400)
         except TypeError:
             return JsonResponse({'MESSAGE': 'TYPE_ERROR'}, status=400)
+
+class FindUsernameView(View):
+
+    def post(self, request):
+        try:
+            data  = json.loads(request.body)
+            name  = data['name']
+            email = data['email']
+            
+            if not utils.validate_email(email):
+                return JsonResponse({'MESSAGE': 'INVALID_EMAIL'}, status=400)
+            
+            forgotten_user = User.objects.filter(name=name, email=email)
+            
+            if not forgotten_user.exists():
+                return JsonResponse({'MESSAGE': 'WRONG_USER'}, status=400)
+
+            forgotten_user = forgotten_user[0]
+            return JsonResponse({
+                "USERNAME"  : forgotten_user.username[:-3]+ "*" * 3,
+                "CREATED_AT": forgotten_user.created_at
+            }, status=200)
+        except json.decoder.JSONDecodeError:
+            return JsonResponse({'MESSAGE': 'JSON_DECODE_ERROR'}, status=400)
+        except KeyError:
+            return JsonResponse({'MESSAGE': 'KEY_ERROR'}, status=400)
+        except TypeError:
+            return JsonResponse({'MESSAGE': 'TYPE_ERROR'}, status=400)
+ 
+class MakeTemporaryPasswordView(View):
+
+    def post(self, request):
+        try:
+            data     = json.loads(request.body)
+            username = data['username']
+            name     = data['name']
+            email    = data['email']
+
+            if not utils.validate_username(username):
+                return JsonResponse({'MESSAGE': 'INVALID_USERNAME'}, status=400)
+
+            if not utils.validate_email(email):
+                return JsonResponse({'MESSAGE': 'INVALID_EMAIL'}, status=400)
+
+            forgotten_user = User.objects.filter(
+                username = username, 
+                name     = name, 
+                email    = email
+            )
+            
+            if not forgotten_user.exists():
+                return JsonResponse({'MESSAGE': 'WRONG_USER'}, status=400)
+
+            forgotten_user          = forgotten_user[0]
+            random_password         = token_urlsafe()[:16]
+            forgotten_user.password = bcrypt.hashpw(
+                random_password.encode('utf-8'), 
+                bcrypt.gensalt()
+            ).decode("utf-8")
+            forgotten_user.save()
+            return JsonResponse({'PASSWORD': random_password}, status=201)
```

Comment:
새로운 자원이 생성된 경우가 아니라 업데이트된 경우이므로 `status=200`이 더 정확합니다.
```suggestion
            return JsonResponse({'PASSWORD': random_password}, status=201)
```

Reply:
감사합니다!!!